pkg/servicemanager/manager.go

```diff
@@ -94,12 +95,28 @@ func (m *ServiceManager) Run(ctx context.Context, ready chan<- struct{}, stopped
 	return ctx.Err()
 }
 
-func (m *ServiceManager) asyncRun(ctx context.Context, service Service) (<-chan struct{}, <-chan struct{}) {
+func (m *ServiceManager) asyncRun(ctx context.Context, cancel context.CancelFunc, service Service) (<-chan struct{}, <-chan struct{}) {
 	ready, stopped := make(chan struct{}), make(chan struct{})
 	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				klog.Errorf("%s crashed in panic: %s", service.Name(), r)
+				klog.Error("Stopping MicroShift")
+				cancel()
```

Comment:
Calling `cancel()` here stops the ServiceManager and its child services, but the main RunMicroShift() routine would still be stuck waiting for a SIGTERM, no?

It seems quite messy calling `cancel()` from multiple locations. Is there a better way handling this, e.g. sending SIGTERM to self, so we'll keep a single flow of control?

Reply:
Makes sense, I believe that would work.

Reply:
Done.